### PR TITLE
Fix `@isolated(any)` for distributed actors

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test

**Review by ktoso — APPROVED** [review]
Looks good, thank you!

File: test/Distributed/distributed_actor_isolated_any.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// CHECK-NEXT:    [[ISOLATION:%.*]] = enum $Optional<any Actor>, #Optional.some!enumelt, [[ISOLATION_OBJECT]] : $any Actor
+// CHECK-NEXT:    [[CLOSURE:%.*]] = partial_apply [callee_guaranteed] [isolated_any] [[CLOSURE_FN]]([[ISOLATION]], [[CAPTURE]])
+// CHECK-NEXT:    // function_ref
+// CHECK-NEXT:    [[TAKE_FN:%.*]] = function_ref @$s4test30takeInheritingAsyncIsolatedAny2fnyyyYaYAc_tF
+// CHECK-NEXT:    apply [[TAKE_FN]]([[CLOSURE]])
+// CHECK-NEXT:    destroy_value [[CLOSURE]]
+distributed actor DA {
+  distributed func testDistributedIsolation() {
+    takeInheritingAsyncIsolatedAny {
+      await self.asyncAction()
```
Comment (ktoso) [inline test/Distributed/distributed_actor_isolated_any.swift:33]: 👍

**Comment by rjmccall** [comment]
@swift-ci Please test

**Comment by glessard** [comment]
@swift-ci please test macOS platform

**Comment by rjmccall** [comment]
@swift-ci Please test

**Pull request merged**